lib/ladybug/energy_model/space_type.rb

```diff
@@ -0,0 +1,284 @@
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+#
+# (3) Neither the name of the copyright holder nor the names of any contributors
+# may be used to endorse or promote products derived from this software without
+# specific prior written permission from the respective party.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE
+# UNITED STATES GOVERNMENT, OR THE UNITED STATES DEPARTMENT OF ENERGY, NOR ANY OF
+# THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+# OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *******************************************************************************
+
+require 'ladybug/energy_model/extension'
+require 'ladybug/energy_model/model_object'
+
+module Ladybug
+  module EnergyModel
+    class SpaceType < ModelObject
+      attr_reader :errors, :warnings
+  
+      def initialize(hash = {})
+        super(hash)
+    
+        raise "Incorrect model type '#{@type}'" unless @type == 'ProgramTypeAbridged'
+      end
+    
+      def defaults
+        result = {}
+        result
+      end
+    
+      def find_existing_openstudio_object(openstudio_model)
+        model_space_type = openstudio_model.getSpaceTypeByName(@hash[:name])
+        return model_space_type.get unless model_space_type.empty?
+        nil
+      end
+    
+      def create_openstudio_object(openstudio_model)    
+        openstudio_space_type = OpenStudio::Model::SpaceType.new(openstudio_model)
+        openstudio_space_type.setName(@hash[:name])
+
+        if @hash[:people]
+          openstudio_people_definition = OpenStudio::Model::PeopleDefinition.new(openstudio_model)
+          openstudio_people_definition.setName(@hash[:people][:name])
+          openstudio_people_definition.setPeopleperSpaceFloorArea(@hash[:people][:people_per_area])
+          if @hash[:people][:radiant_fraction]
+            openstudio_people_definition.setFractionRadiant(@hash[:people][:radiant_fraction])
+          else
+            openstudio_people_definition.setFractionRadiant(@@schema[:definitions][:PeopleAbridged][:radiant_fraction][:default])
+          end
+          #TODO if @hash[:people][:latent_fraction]
+          
+          openstudio_people = OpenStudio::Model::People.new(openstudio_people_definition)
```

Comment:
I think you will want to pull out the creation of People, Lights, Equipment, etc. into their own modules. The way you have it now, this code is going to be duplicated for when you have a Room with a custom People definition that overrides the the Space Type.

Reply:
@chriswmackey I realise what I did. I am assigning the space types but not creating the individual modules for People, Lights, etc at the room level.  I had that in the schema correctly, but have not implemented in the measure.
I will update this for the measure as well.